src/api-service/__app__/onefuzzlib/pools.py

```diff
@@ -258,10 +258,11 @@ def is_outdated(self) -> bool:
         return self.version != __version__
 
     def send_message(self, message: NodeCommand) -> None:
-        NodeMessage(
-            agent_id=self.machine_id,
-            message=message,
-        ).save()
+        if not self.delete_requested:
```

Comment:
We still need to send the messages here.  In the case we're halting the node, we need to tell it to stop processing any tasks immediately and report back that it has done so.